app.py

```diff
@@ -1,56 +1,88 @@
 import datetime
 
 import requests
-from flask import Flask, render_template
+from flask import Flask, render_template, request
 from flask_caching import Cache
 from htmlmin.minify import html_minify
+from constants import CACHE_1_ZIP, CACHE_2_ZIP, CACHE_3_ZIP
+import re
+
+cache = Cache(config=
+    {
+        'CACHE_TYPE': 'simple',
+        'CACHE_DEFAULT_TIMEOUT': 1800
+    })
 
 app = Flask(__name__)
 
-cache = Cache(config={'CACHE_TYPE': 'simple'})
 cache.init_app(app)
 
-
 @app.route('/')
 @app.route('/index')
-@cache.cached(timeout=300)  # 1800 seconds == 30 minutes
-def index():
-
-    timestamp = datetime.datetime.now()
-
-    # // TODO -- MAKE THIS ACCEPT ANY LAT LONG AS A INPUT
-    # latitude and Longitude and  of the area
-    # NYC KNYC central park weather station is Lat: 40.78°N Lon: 73.97°W
-    # long is not negative here because "TOP" in the request accounts for this.
-
-    # Fetching weather data
-    # Hardcoding to the correct weather station
-    weather_data = requests.get(
-        f'https://api.weather.gov/gridpoints/OKX/33,35/forecast'
-    ).json()
-
-    # Period wise data
-    periodic_data = weather_data['properties']['periods']
-
-    latitude, longitude = weather_data['geometry']['geometries'][0]['coordinates'][
-        0], weather_data['geometry']['geometries'][0]['coordinates'][1]
-
-    # Fetching area data
-    area_data = requests.get(
-        f'https://api.weather.gov/points/{longitude},{latitude}'
-    ).json()
-
-    print(area_data)
-
-    areaDescription = area_data['properties']['relativeLocation']['properties']['city'] + ', ' + area_data['properties']['relativeLocation']['properties']['state']
-
-    context = {
-        'timestamp': timestamp,
-        'periodic_data': periodic_data,
-        'areaDescription': areaDescription
-    }
-
-    return render_template('index.html', **context)
+@app.route('/<zip_code>')
+def index(zip_code=None):
+    
+    if re.search(r'.*(\d{5}(\-\d{4})?)$', zip_code) or re.search(r'^$', zip_code):
+        if cache.get(CACHE_1_ZIP) is not None and cache.get(CACHE_1_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_1_ZIP)['context'])
+        elif cache.get(CACHE_2_ZIP) is not None and cache.get(CACHE_2_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_2_ZIP)['context'])
+        elif cache.get(CACHE_3_ZIP)  is not None and cache.get(CACHE_3_ZIP)['zip_code'] == zip_code:
+            return render_template('index.html', **cache.get(CACHE_3_ZIP)['context'])
+        else:
+            timestamp = datetime.datetime.now()
+
+            lat, lng, default_value = 0, 0, False
+
+            if zip_code is None:
+                zip_code, default = 10019, True #New York, Manhattan, Central Park (default)
+
+            with open('./assets/data.csv', 'r') as f:
```

Comment:
where did you source the zipcodes and lat/long? Commenting with the source or how to verify would be useful to troubleshoot in the future. I also recommend renaming `data.csv` to something more semantic to reflect *what*  the data is. For example: `lat-long-zipcode.csv`

Reply:
I'm going to test this locally soon

Reply:
> 
> 
> where did you source the zipcodes and lat/long? Commenting with the source or how to verify would be useful to troubleshoot in the future. I also recommend renaming `data.csv` to something more semantic to reflect _what_ the data is. For example: `lat-long-zipcode.csv`

Sure! Thanks for the advice, gonna change it right away!

Reply:
Sources are from here https://gist.github.com/abatko/ee7b24db82a6f50cfce02afafa1dfd1e